src/registrar/management/commands/utility/extra_transition_domain_helper.py

```diff
@@ -0,0 +1,1022 @@
+
+            Each file_type is a dictionary which
+            contains a dictionary of row[id_field]: object.
+
+            In practice, this would look like:
+
+            EnumFilenames.AUTHORITY_ADHOC: {
+                "1": AuthorityAdhoc(...),
+                "2": AuthorityAdhoc(...),
+                ...
+            }
+
+            desired_id will then specify which id to grab. If we wanted "1",
+            then this function will return the value of id "1".
+            So, `AuthorityAdhoc(...)`
+        """
+        # Grabs a dict associated with the file_type.
+        # For example, EnumFilenames.DOMAIN_ADDITIONAL.
+        desired_type = self.parsed_data_container.file_data.get(file_type)
+        if desired_type is None:
+            self.parse_logs.create_log_item(
+                file_type,
+                LogCode.ERROR,
+                f"Type {file_type} does not exist",
+            )
+            return None
+
+        # Grab the value given an Id within that file_type dict.
+        # For example, "igorville.gov".
+        obj = desired_type.data.get(desired_id)
+        if obj is None:
+            self.parse_logs.create_log_item(
+                file_type,
+                LogCode.ERROR,
+                f"Id {desired_id} does not exist for {file_type.value[0]}",
+            )
+        return obj
+
+
+# TODO - change name
+@dataclass
+class FileDataHolder:
+    """Helper class that holds data about a requested file.
+
+    filename: str -> The desired filename to target. If no filename is given,
+    it is assumed that you are passing in a filename pattern and it will look
+    for a filename that matches the given postfix you pass in.
+
+    regex: re.Pattern -> Defines what regex you want to use when inferring
+    filenames. If none, no matching occurs.
+
+    data_type: type -> Metadata about the desired type for data.
+
+    id_field: str -> Defines which field should act as the id in data.
+    This is necessary as we store lists of "data_type" in ExtraTransitionDomain as follows:
+        {
+            id_field: data_type(...),
+            id_field: data_type(...),
+            ...
+        }
+    """  # noqa
+
+    def __init__(
+        self,
+        filename: str,
+        regex: re.Pattern,
+        data_type: type,
+        id_field: str,
+    ):
+        # Metadata #
+        # = Filename inference metadata =#
+        self.regex = regex
+        self.could_infer = False
+
+        # = "data" object metadata =#
+        # == Where the data is sourced from ==#
+        self.filename = filename
+
+        # == What type the data is ==#
+        self.data_type = data_type
+
+        # == What the id should be in the holding dict ==#
+        # TODO - rename to id_field_name
+        self.id_field = id_field
+
+        # Object data #
+        self.data: Dict[str, type] = {}
+
+    def try_infer_filename(self, current_file_name, default_file_name):
```

Comment:
**Comment (blocking)**: We should never infer filenames. The user who constructs migration_filepaths.json must give the right ones. This is too important to guess. Please pull all of this out. No regexes, no dates, just give the right filename in the JSON input.

Reply:
The inference stuff is a tool for debugging / local development but not used on the script normally (so I can hot swap old data for the latest). This only occurs when the flag --infer_filenames is passed in which is defaulted to false. With that optional bool off, it will throw an error if it can't find the filename normally.
 
 Should this still be removed? In practice this action will never occur unless specified. I can also just hard code it such that the flag is always false and not give the end user the option to even pass it in, which is fine since its just for development. I personally would opt for that approach.

Reply:
**comment**: I'm fine with only doing this in DEBUG environments, that feels very safe.